File: backend/internal/models/notification.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Notification represents an Android notification in the system
type Notification struct {
	gorm.Model
	Title       string    `json:"title" gorm:"not null"`
	Message     string    `json:"message" gorm:"not null"`
	Timestamp   time.Time `json:"timestamp" gorm:"not null;index"`
	PackageName string    `json:"package_name" gorm:"not null;index"`
	From        string    `json:"from" gorm:"index"`
	DeviceID    string    `json:"device_id" gorm:"not null;index"`
} 

File: app/src/main/java/com/sgzmd/lileye/model/Notification.kt
package com.sgzmd.lileye.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST
import java.time.Instant

@Parcelize
data class Notification(
    val packageName: String,
    val title: String?,
    val text: String?,
    val timestamp: Instant,
    val extras: Map<String, String> = emptyMap(),
    val deviceId: String,
) : Parcelable

interface NotificationApi {
    @POST("/api/notifications")
    suspend fun createNotification(@Body notification: Notification): Response<Void>
}